src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllProvider.cs

```diff
@@ -68,6 +100,14 @@ public CallbackDocumentBasedFixAllProvider(Func<FixAllContext, Document, Immutab
                 _fixAllAsync = fixAllAsync;
```

Comment:
my *string* preference is that constructors chain horizontally in a type (e.g. use `this`), then a single constructor uses `base` to walk up the inheritance hierarchy.  That way any special logic (like assigning fields) is always only in one constructor.

Reply:
Ideally, yes. But we don't have access to `GetDefaultSupportedFixAllScopes` API here, so we need this ctor to call into the parameterless public ctor of `DocumentBasedFixAllProvider`